### PR TITLE
Add option for use system libappimage

### PR DESCRIPTION
**Pull request by smarttowel**

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 add_subdirectory(zsync2 EXCLUDE_FROM_ALL)
 
-if(NOT TARGET libappimage)
+if(NOT USE_SYSTEM_LIBAPPIMAGE)
     add_subdirectory(libappimage EXCLUDE_FROM_ALL)
-endif()
+endif()
```
Comment (TheAssassin) [inline lib/CMakeLists.txt:5]: Blank line removed at end of file for no apparent reason.

**Review by TheAssassin — CHANGES_REQUESTED** [review]
You'd have to make sure libappimage is installed in some way.

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,11 @@
 add_subdirectory(zsync2 EXCLUDE_FROM_ALL)
 
-if(NOT TARGET libappimage)
+if(NOT USE_SYSTEM_LIBAPPIMAGE)
     add_subdirectory(libappimage EXCLUDE_FROM_ALL)
+    set(APPIMAGE_LIB libappimage_shared PARENT_SCOPE)
+else()
+    find_library(APPIMAGE_LIB appimage)
```
Comment (TheAssassin) [inline lib/CMakeLists.txt:7]: You should be able to find libappimage using `find_package`, since it generates an exported targets config.

**Review by TheAssassin — COMMENTED** [review]
I am reworking the dependencies management at the moment. I will either merge your PR in the process, or write my own fix. libappimage is not the only external library, same goes for zsync2 for instance.

Thanks for bringing this up.

**Comment by smarttowel** [comment]
I agree about similar options for other external libraries, but I need only system libappimage at this moment.

**Comment by TheAssassin** [comment]
Fixed in https://github.com/AppImage/AppImageUpdate/commit/7593d5aa9c1adfb450e79113fdb6e2f245233cf4. Just set `-DUSE_SYSTEM_LIBAPPIMAGE=ON`.

**Pull request closed without merging**